File: android/app/src/main/java/com/phicdy/totoanticipation/model/RakutenTotoRequestExecutor.java
package com.phicdy.totoanticipation.model;

import android.support.annotation.NonNull;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class RakutenTotoRequestExecutor {

    public interface RakutenTotoRequestCallback {
        void onResponseTotoTop(@NonNull Response<ResponseBody> response);
        void onFailureTotoTop(Call<ResponseBody> call, Throwable throwable);
        void onResponseTotoInfo(@NonNull Response<ResponseBody> response);
        void onFailureTotoInfo(Call<ResponseBody> call, Throwable throwable);
    }
    private final RakutenTotoService service;
    public RakutenTotoRequestExecutor(@NonNull RakutenTotoService service) {
        this.service = service;
    }

    public void fetchRakutenTotoTopPage(@NonNull final RakutenTotoRequestCallback callback) {
        Callback<ResponseBody> topPageCallback = new Callback<ResponseBody>() {
            @Override
            public void onResponse(Call<ResponseBody> call, retrofit2.Response<ResponseBody> response) {
                callback.onResponseTotoTop(response);
            }

            @Override
            public void onFailure(Call<ResponseBody> call, Throwable throwable) {
                callback.onFailureTotoTop(call, throwable);
            }
        };
        Call<ResponseBody> call = service.schedule();
        call.enqueue(topPageCallback);
    }

    public void fetchRakutenTotoInfoPage(@NonNull String num,
                                         @NonNull final RakutenTotoRequestCallback callback) {
        Callback<ResponseBody> rakutenTotoInfoPageCallback = new Callback<ResponseBody>() {
            @Override
            public void onResponse(Call<ResponseBody> call, retrofit2.Response<ResponseBody> response) {
                callback.onResponseTotoInfo(response);
            }

            @Override
            public void onFailure(Call<ResponseBody> call, Throwable throwable) {
                callback.onFailureTotoInfo(call, throwable);
            }
        };
        Call<ResponseBody> call = service.totoInfo(num);
        call.enqueue(rakutenTotoInfoPageCallback);
    }

}

File: android/app/src/main/java/com/phicdy/totoanticipation/model/RakutenTotoRequestExecutor.kt
package com.phicdy.totoanticipation.model

import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class RakutenTotoRequestExecutor(private val service: RakutenTotoService) {

    interface RakutenTotoRequestCallback {
        fun onResponseTotoTop(response: Response<ResponseBody>)
        fun onFailureTotoTop(call: Call<ResponseBody>, throwable: Throwable)
        fun onResponseTotoInfo(response: Response<ResponseBody>)
        fun onFailureTotoInfo(call: Call<ResponseBody>, throwable: Throwable)
    }

    fun fetchRakutenTotoTopPage(callback: RakutenTotoRequestCallback) {
        val topPageCallback = object : Callback<ResponseBody> {
            override fun onResponse(call: Call<ResponseBody>, response: retrofit2.Response<ResponseBody>) {
                callback.onResponseTotoTop(response)
            }

            override fun onFailure(call: Call<ResponseBody>, throwable: Throwable) {
                callback.onFailureTotoTop(call, throwable)
            }
        }
        val call = service.schedule()
        call.enqueue(topPageCallback)
    }

    fun fetchRakutenTotoInfoPage(num: String,
                                 callback: RakutenTotoRequestCallback) {
        val rakutenTotoInfoPageCallback = object : Callback<ResponseBody> {
            override fun onResponse(call: Call<ResponseBody>, response: retrofit2.Response<ResponseBody>) {
                callback.onResponseTotoInfo(response)
            }

            override fun onFailure(call: Call<ResponseBody>, throwable: Throwable) {
                callback.onFailureTotoInfo(call, throwable)
            }
        }
        val call = service.totoInfo(num)
        call.enqueue(rakutenTotoInfoPageCallback)
    }

}
